### PR TITLE
Add an LTC fitting tool and supporting all the BRDFs available in HDRP.

### PR DESCRIPTION
**Pull request by anisunity**
This PR adds a tool to generate the LTC tables for various lighting models (including the possibility for users to generate theirs). The code is greatly inspired by this https://github.com/selfshadow/ltc_code

The tool can be found in Window - > Rendering -> HDRP LTC Tables Generator

Most of the LTC tests do not fail as a managed to generate tables that are extremely close to the previous ones

Here is the list of the current set of brdfs that are "supported"
        GGX,
        DisneyDiffuse,
        Charlie,
        FabricLambert,
        KajiyaKaySpecular,
        KajiyaKayDiffuse,
        CookTorrance,
        Ward,
        OrenNayar,

image

Testing Status
All DXR tests are green locally.
All HDRP tests are green except for the following:
- The two new tests that were added in this PR
- AxF_SVBRDF that fails for me on master aswell.
- Enlighten Area light that fails for epsilon changes.

@remi-chapelain A new tests for area lights and AxF needs to be added.

### CODE REVIEW

**Comment by anisunity** [comment]
@JordanL8 i guess we need docs for this somewhere

File: com.unity.render-pipelines.high-definition/Editor/Material/LTCAreaLight/LTCTableGeneratorEditor.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            {
+                BRDFGeneratorArray = new LTCTableGenerator.BRDFGenerator[brdfTypes.Length];
+                for (int i = 0; i < brdfTypes.Length; ++i)
+                {
+                    BRDFGeneratorArray[i] = new LTCTableGenerator.BRDFGenerator(brdfTypes[i], k_TableResolution, k_DefaultSampleCount, LTCTableGenerator.LTCTableParametrization.CosTheta, k_OutputDirectory);
+                }
+            }
+        }
+
+        [MenuItem("Window/Rendering/HDRP LTC Tables Generator")]
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Editor/Material/LTCAreaLight/LTCTableGeneratorEditor.cs:58]: check with Alex for naming

**Review by sebastienlagarde — APPROVED** [review]

**Comment by remi-chapelain** [comment]
Note for @Vic-Cooper to remove those limitations from theses 5 pages in the doc : 
- [Cotton/Wool Shader](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@10.4/manual/cotton-wool-shader.html#limitations)
- [Silk Shader](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@10.4/manual/silk-shader.html#limitations)
- [Hair Shader](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@10.4/manual/hair-shader.html#limitations)
- [Fabric Master Stack](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@10.4/manual/master-stack-fabric.html#limitations)
- [Hair Master Stack](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@10.4/manual/master-stack-hair.html#limitations)

**Review by remi-chapelain — COMMENTED** [review]
Haven't finished testing but I'm not sure it's expected to be able to put negative value in the sample count field
image

**Comment by sebastienlagarde** [comment]
> Haven't finished testing but I'm not sure it's expected to be able to put negative value in the sample count field
> image

Not expected, but not a good deal either. I will fix it

**Comment by sebastienlagarde** [comment]
I have updated the doc, the screenshots and disable the access to the tool, we are not ready yet to expose it (it is useless for users to regenerate table in state, we are using it internally only for now). PR is ready

**Review by remi-chapelain — APPROVED** [review]
Tested multiple projects with area lights, found nothing unexpected. 
Tested that Hair / Fabric are now properly supported with area light
Tested area light rectangle shadow. 

Duplicated the 1451_AxF test to add area lights for coverage
1451_AxF_SVBRDF_AreaLights

Modified the 2402_Area_Rect_Shadows to add materials (fabric, hair, eye, axf) to test that they are properly receiving shadows

2402_Area_Rect_Shadow

[Relaunched yamato](https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/HDRP%252Fltc-fitting-brdf/.yamato%252Fall-hdrp.yml%2523PR_HDRP_trunk/6004213/job) to regenerate those screenshots.

File: com.unity.render-pipelines.high-definition/Documentation~/whats-new-12.md [hunk 2]
```diff
@@ -15,6 +15,10 @@ Group of Materials / GameObject can be setup to use Force Emissive forward with
 
 ## Improvements
 
+### Area Lights
+
+Area lights are now correctly supported for Fabric and Hair master node as well as for AxF shader.
```
Comment (Vic-Cooper) [inline com.unity.render-pipelines.high-definition/Documentation~/whats-new-12.md:20]: The AxF shader and Fabric and Hair master nodes now correctly support Area lights.

File: com.unity.render-pipelines.high-definition/CHANGELOG.md [hunk 3]
```diff
@@ -35,6 +35,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Added a "Conservative" mode for shader graph depth offset.
 - Added an error message when trying to use disk lights with realtime GI (case 1317808).
 - Added support for multi volumetric cloud shadows.
+- Added an LTC Fitting tools for all the BRDFs supported by HDRP.
+- Adding the support for area lights for Hair and Fabric.
+- using the proper lighting models for AxF for area lights.
```
Comment (Vic-Cooper) [inline com.unity.render-pipelines.high-definition/CHANGELOG.md:41]: - Added LTC Fitting tools for all BRDFs that HDRP supports.
- Added Area Light support for Hair and Fabric master nodes.
- Improved lighting models for AxF shader area lights.

**Review by Vic-Cooper — CHANGES_REQUESTED** [review]
Minor adjustments to the whats new and changelog pages. Thanks for removing the limitations :)

**Comment by sebastienlagarde** [comment]
doc feedback adressed

**Review by Vic-Cooper — APPROVED** [review]
Docs changes addressed - approved!

**Pull request merged**